src/utilities/findDeprecatedUsages.js

```diff
@@ -0,0 +1,60 @@
+/* @flow */
+/**
+ *  Copyright (c) Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import { GraphQLError } from '../error/GraphQLError';
+import { visit, visitWithTypeInfo } from '../language/visitor';
+import type { DocumentNode } from '../language/ast';
+import { getNamedType } from '../type/definition';
+import { GraphQLSchema } from '../type/schema';
+import { TypeInfo } from './TypeInfo';
+
+/**
+ * A validation rule which reports deprecated usages.
+ *
+ * Returns a list of GraphQLError instances describing every deprecated use.
+ */
+export function findDeprecatedUsages(
+  schema: GraphQLSchema,
+  ast: DocumentNode,
+): Array<GraphQLError> {
+  const errors = [];
+  const typeInfo = new TypeInfo(schema);
+
+  visit(ast, visitWithTypeInfo(typeInfo, {
+    Field(node) {
+      const fieldDef = typeInfo.getFieldDef();
+      if (fieldDef && fieldDef.isDeprecated) {
+        const parentType = typeInfo.getParentType();
+        if (parentType) {
+          errors.push(new GraphQLError(
+            `The field ${parentType.name}.${fieldDef.name} is deprecated. ` +
+            (fieldDef.deprecationReason || ''),
```

Comment:
#ocd

Would be nice to avoid the trailing whitespace when there is no `deprecationReason`. Maybe:

```
if (parentType) {
  const reason = fieldDef.deprecationReason;
  errors.push(new GraphQLError(
    `The field ${parentType.name}.${fieldDef.name} is deprecated.` +
    (reason ? ` ${reason}` || ''),
    ...
```

Or similar.

(And ditto below).